media/components/sdfv/vscode_sdfv.js

```diff
@@ -1,3 +1,816 @@
+        });
+
+        s.nodes.forEach((v) => {
+            if (v.attributes.layout)
+                delete v.attributes.layout;
+
+            if (v.type === 'NestedSDFG')
+                un_graphiphy_sdfg(v.attributes.sdfg);
+        });
+    });
+}
+
+function vscode_write_graph(g) {
+    un_graphiphy_sdfg(g);
+    if (vscode)
+        vscode.postMessage({
+            type: 'dace.write_edit_to_sdfg',
+            sdfg: JSON.stringify(g),
+        });
+}
+
+function get_element_metadata(elem) {
+    let metadata = undefined;
+    if (window.sdfg_meta_dict) {
+        if (elem.data) {
+            if (elem.data.sdfg) {
+                metadata = window.sdfg_meta_dict[elem.data.sdfg.type];
+            } else if (elem.data.state) {
+                metadata = window.sdfg_meta_dict[elem.data.state.type];
+            } else if (elem.data.node) {
+                const node_type = elem.data.node.type;
+                if (elem instanceof ScopeNode) {
+                    let node_meta = window.sdfg_meta_dict[node_type];
+                    let scope_meta = undefined;
+                    let entry_idx = node_type.indexOf('Entry');
+                    let exit_idx = node_type.indexOf('Exit');
+                    if (entry_idx)
+                        scope_meta = window.sdfg_meta_dict[
+                            node_type.substring(0, entry_idx)
+                        ];
+                    else if (exit_idx)
+                        scope_meta = window.sdfg_meta_dict[
+                            node_type.substring(0, exit_idx)
+                        ];
+
+                    metadata = {};
+                    if (node_meta !== undefined)
+                        Object.keys(node_meta).forEach(k => {
+                            metadata[k] = node_meta[k];
+                        });
+                    if (scope_meta !== undefined)
+                        Object.keys(scope_meta).forEach(k => {
+                            metadata[k] = scope_meta[k];
+                        });
+                } else if (node_type === 'LibraryNode') {
+                    metadata = window.sdfg_meta_dict[
+                        elem.data.node.classpath
+                    ];
+                } else {
+                    metadata = window.sdfg_meta_dict[node_type];
+                }
+            } else if (elem.data.type) {
+                metadata = window.sdfg_meta_dict[elem.data.type];
+            }
+        } else if (elem.type) {
+            metadata = window.sdfg_meta_dict[elem.type];
+        }
+    } else {
+        // If SDFG property metadata isn't available, query it from DaCe.
+        vscode.postMessage({
+            type: 'dace.query_sdfg_metadata',
+        });
+        // TODO: This needs to be handled more gracefully.
+    }
+    return metadata;
+}
+
+function generate_attributes_table(elem, attributes, root) {
+    let metadata = get_element_metadata(elem);
+
+    for (const attr of attributes) {
```

Comment:
I'd consider refactoring loop contents (after the `if(attr...` part) to a separate function (see dict comment below)